libs/ledger-live-common/src/families/evm/api/rpc.common.ts

```diff
@@ -79,6 +82,20 @@ export const getBalance = (
     return new BigNumber(balance.toString());
   });
 
+/**
+ * Get the balance of an address
+ */
+export const getTokenBalance = (
+  currency: CryptoCurrency,
+  address: string,
+  contractAddress: string
+): Promise<BigNumber> =>
+  withApi(currency, async (api) => {
```

Comment:
the function is called withAPI but returns a jsonRpcProvider, shouldn't it be called `withProvider` ?

Reply:
It's mostly to follow the developer portal doc as that's the [recommended name](https://developers.ledger.com/docs/coin/light-sync/#api-example) for calling external endpoints for a sync.
Also, even if I agree with you that it could be better looking aestically, we're still technically using an API as `ethers` is abstracting the JSON RPC part for us (which would still be a JSON RPC API 😝) 🤷